

{% include install_instructions/shell.html %}
{% include install_instructions/editor.html %}

{% if site.flavor == "r" %}
{% include install_instructions/r.html %}

{% elsif site.flavor == "FIXME" %}
{% include install_instructions/r.html %}
{% else %}
{% include warning-flavor.html %}
{% endif %}

{% comment %}
The following setup instructions are commented out because Carpentries workshops
cover the these topics less frequently. Please uncomment the lines that
correspond to the topics covered in your workshop.

{% include install_instructions/sql.html %}
{% include install_instructions/openrefine.html %}
{% endcomment %}


### Required  software  
If you are not in a Data Carpentry Workshop, the software you need is listed in the table below. Follow the instructions in Option A *or* Option B to have access to these programs.  

| Software website | Used Version in Conda| Manual | Available for | Description |
| -------- | ------------ | ------ | ------------- | ----------- |
| [FastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/) | [0.11.9](https://anaconda.org/bioconda/fastqc) | [Help](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/)| Linux, macOS, Windows | Quality control tool for high throughput sequence data.|
| [Trimmomatic](http://www.usadellab.org/cms/?page=trimmomatic) | [0.39](https://anaconda.org/bioconda/trimmomatic) | [GitHub](https://github.com/usadellab/Trimmomatic) | Linux, macOS, Windows | A flexible read trimming tool for Illumina NGS data. |
|[Kraken](http://ccb.jhu.edu/software/kraken2/)| [2.1.2](https://anaconda.org/bioconda/kraken2)|[GitHub](https://github.com/DerrickWood/kraken2/wiki/Manual)|Linux, macOS|A tool for taxonomic assignation for reads from metagenomics|
|[KronaTools](https://github.com/marbl/Krona/tree/master/KronaTools) |[2.8.1](https://anaconda.org/bioconda/krona)|[GitHub](https://github.com/marbl/Krona/wiki/KronaTools)|Linux, macOS, Windows|A tool for taxonomic visualization in hierarchical pie graphs.|
|[MaxBin2](https://sourceforge.net/projects/maxbin2/)|[2.2.7](https://anaconda.org/bioconda/maxbin2)| [SourceForge](https://sourceforge.net/projects/maxbin2/)|Linux, macOS| Tool for MAGs reconstruction|
|[Spades](https://cab.spbu.ru/software/spades/)|[3.15.2](https://anaconda.org/bioconda/spades)|[GitHub](https://github.com/ablab/spades#sec3)|Linux, macOS| Tool for assemblies|
|[Kraken-biom](https://github.com/smdabdoub/kraken-biom)|[1.2.0](https://anaconda.org/bioconda/kraken-biom)|[GitHub](https://github.com/smdabdoub/kraken-biom)|Linux, macOS, Windows|Tool to convert kraken reports in R readable files|
|[CheckM-genome](https://ecogenomics.github.io/CheckM/)|[1.2.1](https://anaconda.org/bioconda/checkm-genome)|[Wiki](https://github.com/Ecogenomics/CheckM/wiki/Quick-Start#typical-workflow)|Linux, macOs, Windows|Tool to check completeness and contamination in MAGs |  

### Option A: Using the lessons with Amazon Web Services (AWS)

Follow these [instructions on creating an Amazon instance](https://carpentries-lab.github.io/metagenomics-workshop/AMI-setup/index.html). Use the AMI `ami-0f58e878fa70cc201` named `The Carpentries Lab Metagenomics v1.0` listed on the Community AMIs page. Please note that you must set your location as `N. Virginia` to access this community AMI. You can change your location in the upper right corner of the main AWS menu bar. The cost of using this AMI for a few days, with the t2.medium instance type, is very low (about USD $2.00 per user per day). Data Carpentry has *no* control over AWS pricing structure and provides this cost estimate without guarantees. Please read AWS documentation on pricing for up-to-date information. 

If you're an Instructor or Maintainer or want to contribute to these lessons, don't hesitate to contact us at [team@carpentries.org](mailto:team@carpentries.org), and we will start instances for you. 

In this instance, you can use the terminal available in RStudio, and users won't need
to install their terminals or use `ssh` (see [Instructor Notes](https://carpentries-lab.github.io/metagenomics-workshop/guide/index.html)). **If, nevertheless, you
prefer that the users install their own terminals**, directions to install them are included 
for each Windows, Mac OS X, and Linux below in the Option B section. For Windows, you will need to install Git Bash, PuTTY, or the Ubuntu Subsystem.

### Option B: Following the lessons on your local machine  
If you trust that your computer is powerful enough and want to have all the programs installed, you can follow all the workshops without using an
AWS remote machine. To do this, you will need to install all of the software used in the workshop and obtain a copy of the
dataset. Instructions for doing this are below.  


#### **Data**   
The data used in this workshop are available on Zenodo. Please read the Zenodo 
page linked below for information about the data and access to the data files. Because this workshop works 
with real data, be aware that file sizes for the data are large.
[![DOI](https://zenodo.org/badge/DOI/10.5281/zenodo.4285900.svg)](https://doi.org/10.5281/zenodo.4285900)

More information about these data will be presented in the [first episode of the Data processing and visualization for metagenomics lesson](https://carpentries-lab.github.io/metagenomics-analysis/01-background-metadata/index.html).


#### **Install a Bash terminal** 

> ## Windows
> - Download the [Git for Windows installer](https://git-for-windows.github.io/). Run the installer and follow the steps below:
>   + Click on "Next" four times (two times if you've previously installed Git). You don't need to change anything in the information, location, components, and start menu screens.
>   + Select "Use the nano editor by default" and click on "Next".
>   + Keep "Use Git from the Windows Command Prompt" selected and click on "Next". If you forget to do this, the programs that you need for the workshop will not work properly. If this happens, rerun the installer and select the appropriate option.
>   + Select "Use bundled OpenSSH" and click on "Next".
>   + Select "Use the OpenSSL Library" and click "Next".
>   + Keep "Checkout Windows-style, commit Unix-style line endings" selected and click on "Next".
>   + Select "Use Windows' default console window" and click on "Next".
>   + Select "Default (fast-forward on merge)" and click on "Next".
>   + Select "None" (Do not use a credential helper) and click on "Next".
>   + Select "Enable file system caching" and click on "Next".
>   + Ignore "Configuring experimental options" and click on "Install".
>   + Click on "Install".
>   + Click on "Finish".
>   + If your "HOME" environment variable is not set (or you don't know what this is):
>   + Open command prompt (Open Start Menu, then type `cmd` and press [Enter])
>   + Type the following line into the command prompt window exactly as shown: `setx HOME "%USERPROFILE%"`
>   + Press [Enter], and you should see `SUCCESS: Specified value was saved.`
>   + Quit the command prompt by typing `exit` and then pressing [Enter]
>   + See the [video tutorial](https://youtu.be/yo7Z-BEG62A) for an example of how to install Git on Windows 11.
>   
> - An **alternative option** is to install PuTTY
>  by going to the [the installation page](https://www.chiark.greenend.org.uk/~sgtatham/putty/latest.html). For most newer computers, click on putty-64bit-X.XX-installer.msi to download the 64-bit version. If you have an older laptop, you may need to get the 32-bit version putty-X.XX-installer.msi. If you aren't sure whether you need the 64 or 32-bit version, you can check your laptop version by following [the instructions here](https://support.microsoft.com/en-us/help/15056/windows-32-64-bit-faq). Once the installer is downloaded, double-click on it, and PuTTY should install.
> - **Another alternative option** is to use the Windows Subsystem Linux (WSL). This option is available for Windows 10 and Windows 11 - detailed [instructions are available here](https://learn.microsoft.com/en-us/windows/wsl/install).
> See the [video tutorial](https://youtu.be/YoNdTuN-YWk) for an example of how to install WSL with Ubuntu 22.04 on Windows 11.
> 
{: .solution}

> ## macOS
> -  The default shell in some versions of macOS is Bash, and Bash is available in all versions, so no need to install anything. You access Bash from the Terminal Application (found in /Applications/Utilities). See how to open the terminal in the [video tutorial](https://www.youtube.com/watch?v=FuNsWg_VzeQ). You can keep the terminal in your dock for this workshop.
{: .solution}

> ## Linux
>  - The default shell is usually Bash, and there is usually no need to install anything. To see if your default shell is Bash type, echo $SHELL in a terminal and press Enter. If the message printed does not end with `/bash`, then your default is something else, and you can run Bash by typing `bash`.
{: .solution}


#### **Install Miniconda3**

These instructions assume familiarity with the command line and with installation 
in general. There are different operating systems and many different versions
of operating systems and environments, so these may not work on your computer. If an 
installation doesn't work for you, please refer to the user guide for the tool listed in the table above.
If you have difficulties with the installations or find better ways to install things in your operating system, please raise an [Issue](https://github.com/carpentries-lab/metagenomics-analysis/issues) to let us know.

To make a [Conda](https://conda.io/projects/conda/en/latest/index.html) environment, first, you need to install Conda. We recommend installing the [Miniconda3](https://docs.conda.io/en/latest/miniconda.html) version. Miniconda is a package manager that includes Conda and its dependencies and simplifies the installation process. Please first install Miniconda3 (installation instructions below) and then proceed to the installation of the environment.

> ## Linux
> 
> To install miniconda3, see the [video tutorial](https://youtu.be/0PqwShSDH20)
{: .solution}

> ## MacOSX
> In a terminal type:
> ~~~
> $ curl -O https://repo.anaconda.com/miniconda/Miniconda3-latest-MacOSX-x86_64.sh
> $ bash Miniconda3-latest-MacOSX-x86_64.sh
> ~~~
> {: .bash}
> Then, follow the instructions that you are prompted with on the screen to install Miniconda3.
{: .solution}

> ## WSL
> See the video tutorial, [installing Miniconda3 on WSL Ubuntu](https://youtu.be/owQgZoE-GrY)
{: .solution}


<section id="required-software" class="level3" data-number="3.6.1">
<h3 data-number="3.6.1" class="anchored" data-anchor-id="required-software"><span class="header-section-number">3.6.1</span> Required software</h3>
<p>If you are not in a Data Carpentry Workshop, the software you need is listed in the table below. Follow the instructions in Option A <em>or</em> Option B to have access to these programs.</p>
<table class="table">
<thead>
<tr class="header">
<th>Software website</th>
<th>Used Version in Conda</th>
<th>Manual</th>
<th>Available for</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a></td>
<td><a href="https://anaconda.org/bioconda/fastqc">0.11.9</a></td>
<td><a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/">Help</a></td>
<td>Linux, macOS, Windows</td>
<td>Quality control tool for high throughput sequence data.</td>
</tr>
<tr class="even">
<td><a href="http://www.usadellab.org/cms/?page=trimmomatic">Trimmomatic</a></td>
<td><a href="https://anaconda.org/bioconda/trimmomatic">0.39</a></td>
<td><a href="https://github.com/usadellab/Trimmomatic">GitHub</a></td>
<td>Linux, macOS, Windows</td>
<td>A flexible read trimming tool for Illumina NGS data.</td>
</tr>
<tr class="odd">
<td><a href="http://ccb.jhu.edu/software/kraken2/">Kraken</a></td>
<td><a href="https://anaconda.org/bioconda/kraken2">2.1.2</a></td>
<td><a href="https://github.com/DerrickWood/kraken2/wiki/Manual">GitHub</a></td>
<td>Linux, macOS</td>
<td>A tool for taxonomic assignation for reads from metagenomics</td>
</tr>
<tr class="even">
<td><a href="https://github.com/marbl/Krona/tree/master/KronaTools">KronaTools</a></td>
<td><a href="https://anaconda.org/bioconda/krona">2.8.1</a></td>
<td><a href="https://github.com/marbl/Krona/wiki/KronaTools">GitHub</a></td>
<td>Linux, macOS, Windows</td>
<td>A tool for taxonomic visualization in hierarchical pie graphs.</td>
</tr>
<tr class="odd">
<td><a href="https://sourceforge.net/projects/maxbin2/">MaxBin2</a></td>
<td><a href="https://anaconda.org/bioconda/maxbin2">2.2.7</a></td>
<td><a href="https://sourceforge.net/projects/maxbin2/">SourceForge</a></td>
<td>Linux, macOS</td>
<td>Tool for MAGs reconstruction</td>
</tr>
<tr class="even">
<td><a href="https://cab.spbu.ru/software/spades/">Spades</a></td>
<td><a href="https://anaconda.org/bioconda/spades">3.15.2</a></td>
<td><a href="https://github.com/ablab/spades#sec3">GitHub</a></td>
<td>Linux, macOS</td>
<td>Tool for assemblies</td>
</tr>
<tr class="odd">
<td><a href="https://github.com/smdabdoub/kraken-biom">Kraken-biom</a></td>
<td><a href="https://anaconda.org/bioconda/kraken-biom">1.2.0</a></td>
<td><a href="https://github.com/smdabdoub/kraken-biom">GitHub</a></td>
<td>Linux, macOS, Windows</td>
<td>Tool to convert kraken reports in R readable files</td>
</tr>
<tr class="even">
<td><a href="https://ecogenomics.github.io/CheckM/">CheckM-genome</a></td>
<td><a href="https://anaconda.org/bioconda/checkm-genome">1.2.1</a></td>
<td><a href="https://github.com/Ecogenomics/CheckM/wiki/Quick-Start#typical-workflow">Wiki</a></td>
<td>Linux, macOs, Windows</td>
<td>Tool to check completeness and contamination in MAGs</td>
</tr>
</tbody>
</table>
</section>
<section id="option-a-using-the-lessons-with-amazon-web-services-aws" class="level3" data-number="3.6.2">
<h3 data-number="3.6.2" class="anchored" data-anchor-id="option-a-using-the-lessons-with-amazon-web-services-aws"><span class="header-section-number">3.6.2</span> Option A: Using the lessons with Amazon Web Services (AWS)</h3>
<p>Follow these <a href="https://carpentries-lab.github.io/metagenomics-workshop/AMI-setup/index.html">instructions on creating an Amazon instance</a>. Use the AMI <code>ami-0f58e878fa70cc201</code> named <code>The Carpentries Lab Metagenomics v1.0</code> listed on the Community AMIs page. Please note that you must set your location as <code>N. Virginia</code> to access this community AMI. You can change your location in the upper right corner of the main AWS menu bar. The cost of using this AMI for a few days, with the t2.medium instance type, is very low (about USD $2.00 per user per day). Data Carpentry has <em>no</em> control over AWS pricing structure and provides this cost estimate without guarantees. Please read AWS documentation on pricing for up-to-date information.</p>
<p>If you’re an Instructor or Maintainer or want to contribute to these lessons, don’t hesitate to contact us at <a href="mailto:team@carpentries.org">team@carpentries.org</a>, and we will start instances for you.</p>
<p>In this instance, you can use the terminal available in RStudio, and users won’t need to install their terminals or use <code>ssh</code> (see <a href="https://carpentries-lab.github.io/metagenomics-workshop/guide/index.html">Instructor Notes</a>). <strong>If, nevertheless, you prefer that the users install their own terminals</strong>, directions to install them are included for each Windows, Mac OS X, and Linux below in the Option B section. For Windows, you will need to install Git Bash, PuTTY, or the Ubuntu Subsystem.</p>
</section>
<section id="option-b-following-the-lessons-on-your-local-machine" class="level3" data-number="3.6.3">
<h3 data-number="3.6.3" class="anchored"><span class="header-section-number">3.6.3</span> Option B: Following the lessons on your local machine</h3>
<p>If you trust that your computer is powerful enough and want to have all the programs installed, you can follow all the workshops without using an AWS remote machine. To do this, you will need to install all of the software used in the workshop and obtain a copy of the dataset. Instructions for doing this are below.</p>
<section id="data" class="level4" data-number="3.6.3.1">
<h4 data-number="3.6.3.1" class="anchored" data-anchor-id="data"><span class="header-section-number">3.6.3.1</span> <strong>Data</strong></h4>
<p>The data used in this workshop are available on Zenodo. Please read the Zenodo page linked below for information about the data and access to the data files. Because this workshop works with real data, be aware that file sizes for the data are large. <a href="https://doi.org/10.5281/zenodo.4285900"><img src="https://zenodo.org/badge/DOI/10.5281/zenodo.4285900.svg" class="img-fluid" alt="DOI"></a></p>
<p>More information about these data will be presented in the <a href="https://carpentries-lab.github.io/metagenomics-analysis/01-background-metadata/index.html">first episode of the Data processing and visualization for metagenomics lesson</a>.</p>
</section>
<section id="install-a-bash-terminal" class="level4" data-number="3.6.3.2">
<h4 data-number="3.6.3.2" class="anchored"><span class="header-section-number">3.6.3.2</span> <strong>Install a Bash terminal</strong></h4>
<blockquote class="blockquote">
<h2 id="windows" data-number="3.7" class="anchored" data-anchor-id="install-a-bash-terminal"><span class="header-section-number">3.7</span> Windows</h2>
<ul>
<li>Download the <a href="https://git-for-windows.github.io/">Git for Windows installer</a>. Run the installer and follow the steps below:
<ul>
<li>Click on “Next” four times (two times if you’ve previously installed Git). You don’t need to change anything in the information, location, components, and start menu screens.</li>
<li>Select “Use the nano editor by default” and click on “Next”.</li>
<li>Keep “Use Git from the Windows Command Prompt” selected and click on “Next”. If you forget to do this, the programs that you need for the workshop will not work properly. If this happens, rerun the installer and select the appropriate option.</li>
<li>Select “Use bundled OpenSSH” and click on “Next”.</li>
<li>Select “Use the OpenSSL Library” and click “Next”.</li>
<li>Keep “Checkout Windows-style, commit Unix-style line endings” selected and click on “Next”.</li>
<li>Select “Use Windows’ default console window” and click on “Next”.</li>
<li>Select “Default (fast-forward on merge)” and click on “Next”.</li>
<li>Select “None” (Do not use a credential helper) and click on “Next”.</li>
<li>Select “Enable file system caching” and click on “Next”.</li>
<li>Ignore “Configuring experimental options” and click on “Install”.</li>
<li>Click on “Install”.</li>
<li>Click on “Finish”.</li>
<li>If your “HOME” environment variable is not set (or you don’t know what this is):</li>
<li>Open command prompt (Open Start Menu, then type <code>cmd</code> and press [Enter])</li>
<li>Type the following line into the command prompt window exactly as shown: <code>setx HOME "%USERPROFILE%"</code></li>
<li>Press [Enter], and you should see <code>SUCCESS: Specified value was saved.</code></li>
<li>Quit the command prompt by typing <code>exit</code> and then pressing [Enter]</li>
<li>See the <a href="https://youtu.be/yo7Z-BEG62A">video tutorial</a> for an example of how to install Git on Windows 11.</li>
</ul></li>
<li>An <strong>alternative option</strong> is to install PuTTY by going to the <a href="https://www.chiark.greenend.org.uk/~sgtatham/putty/latest.html">the installation page</a>. For most newer computers, click on putty-64bit-X.XX-installer.msi to download the 64-bit version. If you have an older laptop, you may need to get the 32-bit version putty-X.XX-installer.msi. If you aren’t sure whether you need the 64 or 32-bit version, you can check your laptop version by following <a href="https://support.microsoft.com/en-us/help/15056/windows-32-64-bit-faq">the instructions here</a>. Once the installer is downloaded, double-click on it, and PuTTY should install.</li>
<li><strong>Another alternative option</strong> is to use the Windows Subsystem Linux (WSL). This option is available for Windows 10 and Windows 11 - detailed <a href="https://learn.microsoft.com/en-us/windows/wsl/install">instructions are available here</a>. See the <a href="https://youtu.be/YoNdTuN-YWk">video tutorial</a> for an example of how to install WSL with Ubuntu 22.04 on Windows 11.</li>
</ul>
<p>{: .solution}</p>
</blockquote>
<blockquote class="blockquote">
<h2 id="macos" data-number="3.8" class="anchored"><span class="header-section-number">3.8</span> macOS</h2>
<ul>
<li>The default shell in some versions of macOS is Bash, and Bash is available in all versions, so no need to install anything. You access Bash from the Terminal Application (found in /Applications/Utilities). See how to open the terminal in the <a href="https://www.youtube.com/watch?v=FuNsWg_VzeQ">video tutorial</a>. You can keep the terminal in your dock for this workshop. {: .solution}</li>
</ul>
</blockquote>
<blockquote class="blockquote">
<h2 id="linux" data-number="3.9" class="anchored"><span class="header-section-number">3.9</span> Linux</h2>
<ul>
<li>The default shell is usually Bash, and there is usually no need to install anything. To see if your default shell is Bash type, echo $SHELL in a terminal and press Enter. If the message printed does not end with <code>/bash</code>, then your default is something else, and you can run Bash by typing <code>bash</code>. {: .solution}</li>
</ul>
</blockquote>
</section>
<section id="install-miniconda3" class="level4" data-number="3.9.0.1">
<h4 data-number="3.9.0.1" class="anchored"><span class="header-section-number">3.9.0.1</span> <strong>Install Miniconda3</strong></h4>
<p>These instructions assume familiarity with the command line and with installation in general. There are different operating systems and many different versions of operating systems and environments, so these may not work on your computer. If an installation doesn’t work for you, please refer to the user guide for the tool listed in the table above. If you have difficulties with the installations or find better ways to install things in your operating system, please raise an <a href="https://github.com/carpentries-lab/metagenomics-analysis/issues">Issue</a> to let us know.</p>
<p>To make a <a href="https://conda.io/projects/conda/en/latest/index.html">Conda</a> environment, first, you need to install Conda. We recommend installing the <a href="https://docs.conda.io/en/latest/miniconda.html">Miniconda3</a> version. Miniconda is a package manager that includes Conda and its dependencies and simplifies the installation process. Please first install Miniconda3 (installation instructions below) and then proceed to the installation of the environment.</p>
<blockquote class="blockquote">
<h2 id="linux-1" data-number="3.10" class="anchored" data-anchor-id="install-miniconda3"><span class="header-section-number">3.10</span> Linux</h2>
<p>To install miniconda3, see the <a href="https://youtu.be/0PqwShSDH20">video tutorial</a> {: .solution}</p>
</blockquote>
<blockquote class="blockquote">
<h2 id="macosx" data-number="3.11" class="anchored"><span class="header-section-number">3.11</span> MacOSX</h2>
<p>In a terminal type: ~~~ $ curl -O <a href="https://repo.anaconda.com/miniconda/Miniconda3-latest-MacOSX-x86_64.sh" class="uri">https://repo.anaconda.com/miniconda/Miniconda3-latest-MacOSX-x86_64.sh</a> $ bash Miniconda3-latest-MacOSX-x86_64.sh ~~~ {: .bash} Then, follow the instructions that you are prompted with on the screen to install Miniconda3. {: .solution}</p>
</blockquote>
<blockquote class="blockquote">
<h2 id="wsl" data-number="3.12" class="anchored"><span class="header-section-number">3.12</span> WSL</h2>
<p>See the video tutorial, <a href="https://youtu.be/owQgZoE-GrY">installing Miniconda3 on WSL Ubuntu</a> {: .solution}</p>
</blockquote>
</section>
<section id="install-the-metagenomics-environment" class="level4" data-number="3.12.0.1">
<h4 data-number="3.12.0.1" class="anchored"><span class="header-section-number">3.12.0.1</span> <strong>Install the metagenomics environment</strong></h4>
<p>Once your Miniconda3 is ready, follow these instructions to install and activate the metagenomics environment.</p>
<blockquote class="blockquote">
<h2 id="linux-option-1-recommended" data-number="3.13" class="anchored" data-anchor-id="install-the-metagenomics-environment"><span class="header-section-number">3.13</span> Linux: Option 1 (recommended)</h2>
<p>The easier way to install the environment is using the <a href="https://github.com/carpentries-lab/metagenomics-analysis/blob/gh-pages/files/spec-file-Ubuntu22.txt">specifications file</a> for Linux Ubuntu 22.04, which has the exact versions of each tool in this environment. You can use the spec file as follows:<br>
~~~ $ conda create –name metagenomics –file spec-file-Ubuntu22.txt ~~~ {: .bash} More information about how to use environments and spec files is available at <a href="https://conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html">conda documentation</a> {: .solution}</p>
</blockquote>
<blockquote class="blockquote">
<h2 id="linux-option-2" data-number="3.14" class="anchored"><span class="header-section-number">3.14</span> Linux: option 2</h2>
<p>Another way to create an environment is with a <code>ỳml</code> file. This environment can be modified by adding or deleting tools in a file <code>metagenomics-Ubuntu22.yml</code>.</p>
<p>In Ubuntu 22.04, copy this file <a href="https://github.com/carpentries-lab/metagenomics-analysis/blob/gh-pages/files/metagenomics-Ubuntu22.yml">metagenomics-Ubuntu22.yml</a> to your computer and follow the instructions in the <a href="https://youtu.be/0Ks1k9gg3E0">video tutorial</a> {: .solution}</p>
</blockquote>
<blockquote class="blockquote">
<h2 id="macosx-1" data-number="3.15" class="anchored"><span class="header-section-number">3.15</span> MacOSX</h2>
<p>It has been difficult to find compatibility between all the dependencies of each package installed in the metagenomics environment. In the case of the latest version of macOS (Monterey), the MaxBin2 package can be installed, but it does not fully work at the time of use. Copy the file <a href="https://github.com/carpentries-lab/metagenomics-analysis/blob/gh-pages/files/metagenomics-macOS.yml">metagenomics-macOS.yml</a> in your computer and run: ~~~ $ conda env create -f metagenomics-macOS.yml ~~~ {: .bash}<br>
{: .solution}</p>
</blockquote>
<blockquote class="blockquote">
<h2 id="wsl-1" data-number="3.16" class="anchored"><span class="header-section-number">3.16</span> WSL</h2>
<p>In the case of Windows Subsystem for Linux WSL Ubuntu 22.04, the MaxBin2 package has an incompatibility with the checkm-genome package, so we have decided to leave it out of the metagenomics environment and create its own environment with the file(<a href="https://github.com/carpentries-lab/metagenomics-analysis/blob/gh-pages/files/metagenomics-maxbin.yml">metagenomics-maxbin.yml</a>). The file for the metagenomics environment is <a href="https://github.com/carpentries-lab/metagenomics-analysis/blob/gh-pages/files/metagenomics-WSLUbuntu.yml">metagenomics-WSLUbuntu.yml</a> See the <a href="https://youtu.be/kHYmffZYxcI">video tutorial</a> ~~~ $ conda env create -f metagenomics-maxbin.yml $ conda env create -f metagenomics-WSLUbuntu.yml ~~~ {: .bash}<br>
{: .solution}</p>
</blockquote>
</section>
<section id="execute-some-remaining-installation-scripts" class="level4" data-number="3.16.0.1">
<h4 data-number="3.16.0.1" class="anchored" data-anchor-id="execute-some-remaining-installation-scripts"><span class="header-section-number">3.16.0.1</span> <strong>Execute some remaining installation scripts</strong></h4>
<p>Change <code>dcuser</code> with your own username. And run all these lines: ~~~ bash /home/dcuser/.miniconda3/envs/metagenomics/opt/krona/updateTaxonomy.sh<br>
wget <a href="ftp://ftp.ncbi.nih.gov/pub/taxonomy/taxdump.tar.gz" class="uri">ftp://ftp.ncbi.nih.gov/pub/taxonomy/taxdump.tar.gz</a> tar -xzf taxdump.tar.gz mkdir .taxonkit cp names.dmp nodes.dmp delnodes.dmp merged.dmp /home/dcuser/.taxonkit rm *dmp readme.txt taxdump.tar.gz gc.prt ~~~ {: .bash}</p>
</section>
<section id="install-r-and-rstudio" class="level4" data-number="3.16.0.2">
<h4 data-number="3.16.0.2" class="anchored"><span class="header-section-number">3.16.0.2</span> <strong>Install R and RStudio</strong></h4>
<p>R and RStudio are two separate pieces of software:</p>
<p>R is a programming language that is especially powerful for data exploration, visualization, and statistical analysis RStudio is an integrated development environment (IDE) that makes using R easier. In this course, we use RStudio to interact with R.</p>
<blockquote class="blockquote">
<h2 id="mac-os-x" data-number="3.17" class="anchored" data-anchor-id="install-r-and-rstudio"><span class="header-section-number">3.17</span> Mac OS X</h2>
<ul>
<li>Download R from the <a href="https://cran.r-project.org/bin/macosx/">CRAN website</a>.</li>
<li>Select the .pkg file for the latest R version</li>
<li>Double-click on the downloaded file to install R</li>
<li>It is also a good idea to install <a href="https://www.xquartz.org/">XQuartz</a> (needed by some packages)</li>
<li>Go to the <a href="https://www.rstudio.com/products/rstudio/download/#download">RStudio download</a> page</li>
<li>Under Installers, select RStudio x.yy.zzz - Mac OS X 10.6+ (64-bit) (where x, y, and z represent version numbers)</li>
<li>Double-click the file to install RStudio</li>
<li>Once it’s installed, open RStudio to ensure it works and you don’t get any error messages. {: .solution}</li>
</ul>
</blockquote>
<blockquote class="blockquote">
<h2 id="windows-1" data-number="3.18" class="anchored"><span class="header-section-number">3.18</span> Windows</h2>
<ul>
<li>Download R from the <a href="https://cran.r-project.org/bin/windows/base/">CRAN website</a>.</li>
<li>Run the .exe file that was just downloaded</li>
<li>Go to the <a href="https://www.rstudio.com/products/rstudio/download/#download">RStudio download page</a></li>
<li>Under Installers select RStudio x.yy.zzz - Windows Vista/7/8/10 (where x, y, and z represent version numbers)</li>
<li>Double-click the file to install it</li>
<li>Once it’s installed, open RStudio to ensure it works and you don’t get any error messages. {: .solution}</li>
</ul>
</blockquote>
<blockquote class="blockquote">
<h2 id="linux-2" data-number="3.19" class="anchored"><span class="header-section-number">3.19</span> Linux</h2>
<ul>
<li>Follow the instructions for your distribution from <a href="https://cloud.r-project.org/bin/linux">CRAN</a>. They provide information to get the most recent version of R for common distributions. For most distributions, you could use your package manager (e.g., for Debian/Ubuntu, run sudo apt-get install r-base, and for Fedora, sudo yum install R). However, we don’t recommend this approach as the versions provided by this are usually out of date. In any case, make sure you have at least R 3.3.1.</li>
<li>Go to the <a href="https://www.rstudio.com/products/rstudio/download/#download">RStudio</a> download page</li>
<li>Under Installers, select the version that matches your distribution, and install it with your preferred method (e.g., with Debian/Ubuntu ´sudo dpkg -i rstudio-x.yy.zzz-amd64.deb´ at the terminal).</li>
<li>Once it’s installed, open RStudio to ensure it works and you don’t get any error messages. {: .solution}</li>
</ul>
</blockquote>
</section>
<section id="install-r-libraries" class="level4" data-number="3.19.0.1">
<h4 data-number="3.19.0.1" class="anchored" data-anchor-id="install-r-libraries"><span class="header-section-number">3.19.0.1</span> <strong>Install R libraries</strong></h4>
<table class="table">
<thead>
<tr class="header">
<th>Software</th>
<th>Version</th>
<th>Manual</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><a href="https://github.com/joey711/phyloseq">phyloseq</a></td>
<td>1.39.1</td>
<td><a href="https://github.com/joey711/phyloseq">GitHub</a></td>
<td>Explore, manipulate and analyze microbiome profiles with R</td>
</tr>
<tr class="even">
<td><a href="https://ggplot2.tidyverse.org/">ggplot2</a></td>
<td>3.3.6</td>
<td><a href="https://github.com/tidyverse/ggplot2">GitHub</a></td>
<td>System for declaratively creating graphics, based on The Grammar of Graphics</td>
</tr>
</tbody>
</table>
<p>Type these commands in your console: ~~~ &gt; install.packages(“phyloseq”) &gt; install.packages(“ggplot2”) ~~~ {: .language-R}</p>


</section>
</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./data_analysis_methods.html" class="pagination-link" aria-label="Data Analysis Methods">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Data Analysis Methods</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./summary.html" class="pagination-link" aria-label="Summary">
        <span class="nav-page-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Summary</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->




</body></html>
